tools/lz4-compress.js

```diff
@@ -1,157 +1,60 @@
-  }
-
-  function find(filename) {
-    var prefixes = [nodePath.join(__dirname, '..', 'src'), process.cwd()];
-    for (var i = 0; i < prefixes.length; ++i) {
-      var combined = nodePath.join(prefixes[i], filename);
-      if (nodeFS.existsSync(combined)) {
-        return combined;
-      }
-    }
-    return filename;
-  }
-
-  read = function(filename, binary) {
-    filename = nodePath['normalize'](filename);
-    var ret = nodeFS['readFileSync'](filename);
-    // The path is absolute if the normalized version is the same as the resolved.
-    if (!ret && filename != nodePath['resolve'](filename)) {
-      filename = path.join(__dirname, '..', 'src', filename);
-      ret = nodeFS['readFileSync'](filename);
-    }
-    if (ret && !binary) ret = ret.toString();
-    return ret;
-  };
-
-  readBinary = function(filename) { return read(filename, true) };
-
-  load = function(f) {
-    globalEval(read(f));
-  };
-
-  arguments_ = process['argv'].slice(2);
-
-} else if (ENVIRONMENT_IS_SHELL) {
-  // Polyfill over SpiderMonkey/V8 differences
-  if (!this['read']) {
-    this['read'] = function(f) { snarf(f) };
-  }
-
-  if (typeof scriptArgs != 'undefined') {
-    arguments_ = scriptArgs;
-  } else if (typeof arguments != 'undefined') {
-    arguments_ = arguments;
-  }
-
-} else if (ENVIRONMENT_IS_WEB) {
-  this['print'] = printErr = function(x) {
-    console.log(x);
-  };
-
-  this['read'] = function(url) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', url, false);
-    xhr.send(null);
-    return xhr.responseText;
-  };
-
-  if (this['arguments']) {
-    arguments_ = arguments;
-  }
-} else if (ENVIRONMENT_IS_WORKER) {
-  // We can do very little here...
-
-  this['load'] = importScripts;
-
-} else {
-  throw 'Unknown runtime environment. Where are we?';
+function print(x) {
+  process['stdout'].write(x + '\n');
 }
 
-function globalEval(x) {
-  eval.call(null, x);
+function printErr(x) {
+  process['stderr'].write(x + '\n');
 }
 
-if (typeof load === 'undefined' && typeof read != 'undefined') {
-  this['load'] = function(f) {
-    globalEval(read(f));
-  };
+function read(filename, binary) {
+  filename = path['normalize'](filename);
+  let ret = fs['readFileSync'](filename);
+  if (ret && !binary) ret = ret.toString();
+  return ret;
 }
 
-if (typeof printErr === 'undefined') {
-  this['printErr'] = function(){};
+function readBinary(filename) {
+  return read(filename, true);
 }
 
-if (typeof print === 'undefined') {
-  this['print'] = printErr;
+function globalEval(x) {
+  eval.call(null, x);
 }
 
-assert = function(x, message) {
-  if (!x) throw 'assertion failed: ' + message + ' : ' + new Error().stack;
+function load(f) {
+  globalEval(read(f));
 }
 
-if (!Math['imul'] || Math['imul'](0xffffffff, 5) !== -5) Math['imul'] = function imul(a, b) {
-  var ah  = a >>> 16;
-  var al = a & 0xffff;
-  var bh  = b >>> 16;
-  var bl = b & 0xffff;
-  return (al*bl + ((ah*bl + al*bh) << 16))|0;
+assert = function(x, message) {
```

Comment:
Other functions in this file have `function assert(x, message) { ...` form.. Is there a reason why this function is different?

Reply:
Yes, I think its because this function needs to be added to the global namespace so that the script that is `eval` 'ed below had visibility of it.

Its a strange quirk of node and if you just say `foo = 3` that creates a new name `foo` on the global object that is shared. 

I have followup PR that will remove the use of eval completely .. but for now this smaller cleanup still relies on eval and the global shared namespace.